src/components/data-table/ha-data-table.ts

```diff
@@ -192,6 +133,48 @@ export class HaDataTable extends BaseElement {
           break;
         }
       }
+
+      const clonedColumns: DataTabelColumnContainer = deepClone(this.columns);
+      Object.values(clonedColumns).forEach((column: DataTabelColumnData) => {
```

Comment:
Can we memoize this, so we only create the sorted columns once ?

Reply:
We only do it now if the columns are changed, so memoize would do the same?

Reply:
ok!